integration/integration_nexus_upload_test.go

```diff
@@ -0,0 +1,156 @@
+
+	resp, err = http.Get(url + "/repository/maven-releases/com/mycompany/app/my-app/1.0/my-app-1.0.pom")
+	assert.NoError(t, err, "Downloading artifact failed")
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+	resp, err = http.Get(url + "/repository/maven-releases/com/mycompany/app/my-app/1.0/my-app-1.0.jar")
+	assert.NoError(t, err, "Downloading artifact failed")
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+	resp, err = http.Get(url + "/repository/maven-releases/mygroup/mymta/0.3.0/mymta-0.3.0.yaml")
+	assert.NoError(t, err, "Downloading artifact failed")
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+	resp, err = http.Get(url + "/repository/maven-releases/mygroup/mymta/0.3.0/mymta-0.3.0.mtar")
+	assert.NoError(t, err, "Downloading artifact failed")
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+}
+
+func TestNexus2Upload(t *testing.T) {
+	ctx := context.Background()
+	req := testcontainers.ContainerRequest{
+		Image:        "sonatype/nexus:2.14.16-01",
+		ExposedPorts: []string{"8081/tcp"},
+		WaitingFor:   wait.ForLog("org.sonatype.nexus.bootstrap.jetty.JettyServer - Running").WithStartupTimeout(5 * time.Minute), // Nexus takes more than one minute to boot
+	}
+	nexusContainer, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	assert.NoError(t, err)
+	defer nexusContainer.Terminate(ctx)
+	ip, err := nexusContainer.Host(ctx)
+	assert.NoError(t, err)
+	port, err := nexusContainer.MappedPort(ctx, "8081")
+	assert.NoError(t, err, "Could not map port for nexus container")
+	nexusIpAndPort := fmt.Sprintf("%s:%s", ip, port.Port())
+	url := "http://" + nexusIpAndPort + "/nexus/"
+
+	cmd := command.Command{}
+	cmd.SetDir("testdata/TestNexusIntegration/mta")
+
+	piperOptions := []string{
+		"nexusUpload",
+		"--groupId=mygroup",
+		"--artifactId=mymta",
+		"--user=admin",
+		"--password=admin123",
+		"--repository=releases",
+		"--version=nexus2",
+		"--url=" + nexusIpAndPort + "/nexus/",
+	}
+
+	err = cmd.RunExecutable(getPiperExecutable(), piperOptions...)
+	assert.NoError(t, err, "Calling piper with arguments %v failed.", piperOptions)
+
+	cmd = command.Command{}
+	cmd.SetDir("testdata/TestNexusIntegration/maven")
+
+	piperOptions = []string{
+		"nexusUpload",
+		"--user=admin",
+		"--password=admin123",
+		"--repository=releases",
+		"--version=nexus2",
+		"--url=" + nexusIpAndPort + "/nexus/",
+	}
+
+	err = cmd.RunExecutable(getPiperExecutable(), piperOptions...)
+	assert.NoError(t, err, "Calling piper with arguments %v failed.", piperOptions)
+
+	resp, err := http.Get(url + "content/repositories/releases/com/mycompany/app/my-app/1.0/my-app-1.0.pom")
+	assert.NoError(t, err, "Downloading artifact failed")
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+	resp, err = http.Get(url + "content/repositories/releases/com/mycompany/app/my-app/1.0/my-app-1.0.jar")
```

Comment:
do we really need to check in the binary or can we create this on the fly? zip, tar?